File: javapatterns/src/main/java/org/iut/nantes/adapter/canards/Canard.java
package org.iut.nantes.adapter.canards;

public interface Canard {
    void cancaner();
    void voler();
}

class Colvert implements Canard {

    @Override
    public void cancaner() {
        System.out.println("Coin coin.");
    }

    @Override
    public void voler() {
        System.out.println("Je vole.");
    }
}

File: kotlinpatterns/src/main/kotlin/org/iut/nantes/adapter/canards/Canard.kt
package org.iut.nantes.adapter.canards

interface Canard {
    fun cancaner()
    fun voler()
}

internal class Colvert : Canard {
    override fun cancaner() {
        println("Coin coin.")
    }

    override fun voler() {
        println("Je vole.")
    }
}